components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -315,6 +315,44 @@ public void unregisterOAuthApplication(String userId, String applicationName, St
         }
     }
 
+    public RegistrationResponseProfile readOAuthApplication(String consumerKey, String userId)
+        throws DCRException{
+
+        String tenantDomain = MultitenantUtils.getTenantDomain(userId);
+        String userName = MultitenantUtils.getTenantAwareUsername(userId);
+
+        PrivilegedCarbonContext.startTenantFlow();
```

Comment:
Starting tenant flow should also be done within a try block. The finally block of that same try block should end the tenant flow.